apps/userdashboard/assets/js/a4_candy_userdashboard/ModerationNotification.jsx

```diff
@@ -23,35 +23,44 @@ export default class ModerationNotification extends Component {
     )
   }
 
-  toggleIsPending () {
-    const data = { is_pending: !this.state.isPending }
+  safeFetch = (payload) => {
     const headers = {
       Accept: 'application/json',
       'Content-Type': 'application/json',
       'X-CSRFToken': Cookies.get('csrftoken')
     }
-
-    fetch(this.props.apiUrl, {
+    return fetch(this.props.apiUrl, {
       method: 'PATCH',
       headers: headers,
-      body: JSON.stringify(data)
+      body: JSON.stringify(payload)
     })
       .then(res => res.json())
-      .then(json => {
-        this.props.onChangePending(json.is_pending)
-        this.setState({
-          isPending: json.is_pending
-        })
-      }).catch((err) => {
-        console.log(err)
-      })
+      .then(response => [response, undefined])
+      .catch(error => Promise.resolve([undefined, error]))
   }
 
-  toggleIsBlocked () {
-    const newIsBlocked = !this.state.isBlocked
-    this.setState({
-      isBlocked: newIsBlocked
-    })
+  async toggleIsPending () {
+    const [response, error] =
+      await this.safeFetch({ is_pending: !this.state.isPending })
+    const alertMessage = response && response.is_pending
+      ? 'unarchived'
+      : 'archived'
+
+    !error || this.props.onChangeBlocked(error)
+    !error && this.props.onChangeBlocked(alertMessage)
+    !error && this.setState({ isPending: response.is_pending })
+  }
```

Comment:
@khamui can you explain this syntax? i'm not finding it very readable

Reply:
yes i have to admit this is a very shorthand style of writing if statements. I find, if you get use to it, than this allows code to be placed close to each other to make the block even easy to read. Translated into normal if/else notation it would be (I can change that, but i think the extra lines of an if/else is disturbing somehow):

```
if (!error) {
  this.props.onChangeBlocked(alertMessage)
  this.setState({ isPending: response.is_pending })
} else {
  this.props.onChangeBlocked(error)
}
```

this i would need to write two times for `toggleIsPending` and `toggleIsBlocked`. I don't mind to change it...

Reply:
oh the first comment was just the justification haha. the explanation of those lines is:

first part the statement above is the condition `!error` and the second part is the execution `this.props.onChangeBlocked(error)`.

In case of a `&&` operator it will execute if the `!error` is `true`. In case of `||` the second part will execute if `!error` is `false`.

Reply:
I would take readability over less code personally but @Rineee @fuzzylogic2000 what are your thoughts? i'm happy to go with the majority?

Reply:
yes, you are right! i am going to change it, because most probably the both of us are the ones that have to change it in the future. lets go for the more convenient way.